lib/Dialect/Linalg/Transforms/LinalgTransforms.cpp

```diff
@@ -106,3 +111,35 @@ bool mlir::linalg::detail::isProducedByOpOfTypeImpl(
   }
   return false;
 }
+
+LogicalResult
+mlir::linalg::permuteGenericLinalgOp(PatternRewriter &rewriter, Operation *op,
+                                     ArrayRef<unsigned> permutation,
+                                     StringRef linalgMarker) {
+  // If permutation is empty, there is nothing to be done.
+  if (permutation.size() == 0)
+    return failure();
+
+  auto linOp = cast<LinalgOp>(op);
+  auto permutationMap =
+      inversePermutation(AffineMap::getPermutationMap(permutation, rewriter.getContext()));
+  SmallVector<AffineMap, 4> newIndexingMap;
+  auto indexingMaps = linOp.getAttrOfType<ArrayAttr>("indexing_maps").getValue();
+  for (unsigned i = 0, e = linOp.getNumInputsAndOutputs(); i != e; ++i) {
+    AffineMap m = indexingMaps[i].cast<AffineMapAttr>().getValue().compose(permutationMap);
+    newIndexingMap.push_back(m);
+  }
+  auto itTypes = linOp.getAttrOfType<ArrayAttr>("iterator_types").getValue();
+  SmallVector<StringRef, 4> itTypesVector;
+  for (unsigned i = 0, e = itTypes.size(); i != e; ++i)
+    itTypesVector.push_back(itTypes[i].cast<StringAttr>().getValue());
+  applyPermutationToVector(itTypesVector, permutation);
+  op->setAttr("indexing_maps",
+              rewriter.getAffineMapArrayAttr(llvm::makeArrayRef(newIndexingMap)));
```

Comment:
Is makeArrayRef actually needed in either of these cases?

Reply:
No, it wasn't after the changes suggested by Nicolas. Thanks for noticing.